_posts/2016-08-26-babili.md

```diff
@@ -0,0 +1,183 @@
+
+```bash
+$ npm install babili --save-dev
+```
+
+```bash
+# equivalent to 
+# babel src -d lib --presets=babili
+$ babili src -d lib
+```
+
+### Webpack
+
+You can either just use the preset option above with `babel-loader`, or use it seperately with the [babili-webpack-plugin](https://github.com/boopathi/babili-webpack-plugin) (made by [@boopathi](https://github.com/boopathi/), who also works on babili).
+
+```bash
+$ npm install babili-webpack-plugin --save-dev
+```
+
+```js
+// webpack.config.js
+const BabiliPlugin = require("babili-webpack-plugin");
+module.exports = {
+  entry: //...,
+  output: //...,
+  plugins: [
+    new BabiliPlugin(options)
+  ]
+}
+```
+
+We want to have a better story with integration with webpack/bundlers in the near future! Ref [#100](https://github.com/babel/babili/issues/100).
+
+## Pros/Cons
+
+Uglify Pros
+- No change to existing tooling if you are already minifying.
+- Battle-tested/production ready (been around for years, and has wide adoption so there's less bugs/issues).
+- It's fast and outputs small code already.
+
+Uglify Cons
+- Custom parser/tooling, so difficult to output/minify ES2015+ and more.
+- Not modular, no way to create own plugins/minification strategies.
+
+Babili Pros:
+- ES2015+ aware (nothing special needs to be done because we can use the babylon parser) and Babel will update as standards/browsers update.
+- Uses the existing babel toolchain, can consume as a babel preset or standalone.
+- Potential for custom smart transforms for React/Flow, etc.
+- Could use flow/typescript annotations to help with minification.
+- Each minfication step can be split into it's own plugins and plenty of options for customization. Makes it easier to both contribute and to find/submit issues for specific problems. Also people can be free to experiment with their own plugins and get them accepted into core.
+  - For example: [this](https://github.com/babel/babili/tree/master/packages/babel-plugin-transform-minify-booleans) just turns `true` into `!0` which is straightforward to write.
+- Should be an easy transition if people are familiar with transpiling with Babel already.
+
+Babili Cons:
+- We released it early, so there aren't many users yet. Early adopters will have to deal with a tool that isn't as battle-tested as Uglify at first.
+- Right now, the performance is worse/size is worse than Uglify on our benchmark tests. However, this is something we'll be focusing on improving.
+
+TL;DR: Babili should be able to keep up with the ECMAScript standard as new features get added as well as target the environments you need to support. It has a lot of potential: it may not be as production-ready as uglify at the moment since it was just released but as we continue to optimize with more users it should be more than capable.
```

Comment:
It seems like we should standardize on `Uglify` capitalized. We're using a random sampling of both uppercase and lowercase throughout this post.